tests/test_testRepo.py

```diff
@@ -43,13 +43,14 @@ def setUpClass(cls):
         # this has a prohibitive run-time cost at present
         cls.root = makeTestTempDir(TESTDIR)
 
-        dataIds = {
-            "instrument": ["notACam", "dummyCam"],
-            "physical_filter": ["k2020", "l2019"],
-            "visit": [101, 102],
-            "detector": [5]
-        }
-        cls.creatorButler = makeTestRepo(cls.root, dataIds)
+        cls.creatorButler = makeTestRepo(cls.root)
+        addDataIdValue(cls.creatorButler, "instrument", "notACam")
+        addDataIdValue(cls.creatorButler, "instrument", "dummyCam")
+        addDataIdValue(cls.creatorButler, "physical_filter", "k2020", band="k", instrument="notACam")
```

Comment:
As usual, test examples are illuminating. Yet, I have some questions. All the `**related` items being input here are nominally of type string (per the function definition), and I guess the list of existing dimensions to be linked could be as long as desired. So, as a user invoking `addDataIdValue`, how to best decide what is related and what's not? For example, am I just supposed to *know* that adding a `physical_filter` requires linking a `band` and an `instrument`? Or is it not required at all and is just used in this test as a feasible example?

Reply:
Yeah, that's something I don't know how to handle.

Looking at https://github.com/lsst/daf_butler/blob/master/python/lsst/daf/butler/configs/dimensions.yaml#L80, `physical_filter` requires `instrument`, but does not require `band`. As far as I know this isn't documented anywhere, even though you also need to know it for writing a `PipelineTask` (the testing of which is the main use case for this code) or even passing data IDs to `pipetask`.

My intent was that this module could be used with no more knowledge of the dimensions system than you need for `pipetask` or to write a `PipelineTask`. Ideally, _all_ these keywords would be optional, with `_fillRelationships` filling them in as needed. However, I found that `_fillRelationships` would often pick contradictory values (e.g., exposure 101 is associated with camera 1 and with the g3 filter, but g3 belongs to camera 2). I couldn't come up with an algorithm that _can't_ do that, so I opted for putting warnings in the error handler and the documentation about the need for human guidance. It turns out the old code had the same problem (but with no workaround), I just got lucky during testing. 😬 

Any advice would be much appreciated.

Reply:
In https://github.com/lsst/daf_butler/blob/master/python/lsst/daf/butler/configs/dimensions.yaml#L81-L82 you can see that physical_filter requires an instrument but implies a band.

Reply:
I guess what I'm really getting at here is this. To improve the tests, is there some combo of things you could put in `**related` that would cause `addDataIdValue` to break because the inputs are self-inconsistent? Like if you told it band is r and physical_filter is HSC-G, I don't think that should be OK.